pkg/plugininventory/sqlite_inventory.go

```diff
@@ -159,13 +159,26 @@ func (b *SQLiteInventory) GetPluginGroups(filter PluginGroupFilter) ([]*PluginGr
 }
 
 // getPluginsFromDB returns the plugins found in the DB 'inventoryFile' that match the filter
+//
+//nolint:dupl
 func (b *SQLiteInventory) getPluginsFromDB(filter *PluginInventoryFilter) ([]*PluginInventoryEntry, error) {
+	// Check if the inventory file exists.
+	if _, err := os.Stat(b.inventoryFile); os.IsNotExist(err) {
+		return []*PluginInventoryEntry{}, nil
+	}
+
 	db, err := sql.Open("sqlite", b.inventoryFile)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to open the DB at '%s'", b.inventoryFile)
 	}
 	defer db.Close()
 
+	// Return empty data if db connection is not available
+	err = db.Ping()
```

Comment:
I have added to make sure we have valid connection to db.
because db.Open may just validate its arguments without creating a connection
to the database. To verify that the data source name is valid, we need to use Ping.